hypnotoad/cases/tokamak.py

```diff
@@ -298,6 +311,33 @@ def __init__(
                options (self.nonorthogonal_options)
 
         """
+
+        # Take the default settings, then the options keyword, then
+        # any additional keyword arguments
+        self.user_options = self.user_options_factory.create(settings)
+
+        if self.user_options.reverse_current:
+            warnings.warn("Reversing the sign of the poloidal field")
+            psi2D *= -1.0
+            psi1D *= -1.0
+
+        if self.user_options.extrapolate_profiles:
+            # Extend the array out to normalised psi of 1.2
+            # Exclude first point since duplicates last point in core
+            psiSOL = np.linspace(0.0, 0.2 * (psi1D[-1] - psi1D[0]), 50)[1:]
+            psi1D = np.concatenate([psi1D, psiSOL + psi1D[-1]])
+
+            if pressure is not None:
+                # Use an exponential decay for the pressure, based on
+                # the value and gradient at the plasma edge
+                dpdpsi = (pressure[-1] - pressure[-2]) / (psi1D[-1] - psi1D[-2])
+                p0 = pressure[-1]
+                # p = p0 * exp( (psi - psi0) * dpdpsi / p0)
+                pressure = np.concatenate([pressure, p0 * np.exp(psiSOL * dpdpsi / p0)])
+
+            # fpol constant in SOL
+            fpol1D = np.concatenate([fpol1D, np.full(psiSOL.shape, fpol1D[-1])])
```

Comment:
We could use the maximum psi value on the grid here to cover all possible (?) cases. Also, I think we need to calculate dpdpsi before extending psi1D so we get the right dpsi (this is how it was done when the setting was applied in `read_geqdsk()`).
```suggestion
            # Extend the array to outer-most psi on grid
            if pressure is not None:
                dpdpsi = (pressure[-1] - pressure[-2]) / (psi1D[-1] - psi1D[-2])

            psi_increasing = psi1D[-1] > psi1D[0]
            if psi_increasing:
                psi_outer = max(self.user_options.psi_sol, self.user_options.psi_sol_inner)
            else:
                psi_outer = min(self.user_options.psi_sol, self.user_options.psi_sol_inner)
            if (
                (psi_increasing and psi_outer > psi1D[-1])
                or (not psi_increasing and psi_outer < psi1D[-1]
            ):
                # if psi_outer is not beyond the last point of psi1D, no need to extend
                # Exclude first point since duplicates last point in core
                psiSOL = np.linspace(psi1D[-1], psi_outer), 50)[1:]
                psi1D = np.concatenate([psi1D, psiSOL + psi1D[-1]])

                # fpol constant in SOL
                fpol1D = np.concatenate([fpol1D, np.full(psiSOL.shape, fpol1D[-1])])

            if pressure is not None:
                # Use an exponential decay for the pressure, based on
                # the value and gradient at the plasma edge
                p0 = pressure[-1]
                # p = p0 * exp( (psi - psi0) * dpdpsi / p0)
                pressure = np.concatenate([pressure, p0 * np.exp(psiSOL * dpdpsi / p0)])
```